slimt/Annotation.hh

```diff
@@ -250,5 +273,19 @@ class AnnotatedText {
     return std::string_view(text.data() + range.begin, range.size());
   }
 };
+class WordIterator {
+ public:
+  explicit WordIterator(const AnnotatedText &annotated)
+      : annotated_(annotated) {}
+  Range operator*();
+  WordIterator &operator++();
+  WordIterator &operator--();
+
+ private:
+  const AnnotatedText &annotated_;
+  size_t sentence_idx_{};
+  size_t word_idx_{};
```

Comment:
Try very very hard to initialize to a legal value when possible. The rule of thumb is to have the object legal at construction. This one should be set to `0`, not `{}`.